src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/TransitiveNoWarnUtils.cs

```diff
@@ -809,11 +811,21 @@ public NodeWarningProperties GetIntersect(NodeWarningProperties other)
                         thisPackages.TryGetValue(key, out var thisCodes);
```

Comment:
There's an opportunity to improve the existing logic. Since this is an intersection, we don't need `allKeys` - we can iterate over just the keys/values in `thisPackages`.

Reply:
Excellent point!